controllers/gitopsservice_controller.go

```diff
@@ -514,7 +531,67 @@ func (r *ReconcileGitopsService) reconcileDefaultArgoCDInstance(instance *pipeli
 
 		if changed {
 			reqLogger.Info("Reconciling ArgoCD", "Namespace", existingArgoCD.Namespace, "Name", existingArgoCD.Name)
-			err = r.Client.Update(context.TODO(), existingArgoCD)
+			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: existingArgoCD.Name, Namespace: existingArgoCD.Namespace}, existingArgoCD); err != nil {
+					return err
+				}
+				changed := false
+				if existingArgoCD.Spec.ApplicationSet != nil {
+					if existingArgoCD.Spec.ApplicationSet.Resources == nil {
+						existingArgoCD.Spec.ApplicationSet.Resources = defaultArgoCDInstance.Spec.ApplicationSet.Resources
+						changed = true
+					}
+				}
+				if existingArgoCD.Spec.Controller.Resources == nil {
+					existingArgoCD.Spec.Controller.Resources = defaultArgoCDInstance.Spec.Controller.Resources
+					changed = true
+				}
+				if argocdcontroller.UseDex(existingArgoCD) {
+					if existingArgoCD.Spec.SSO != nil && existingArgoCD.Spec.SSO.Provider == argoapp.SSOProviderTypeDex {
+						if existingArgoCD.Spec.SSO.Dex != nil {
+							if existingArgoCD.Spec.SSO.Dex.Resources == nil {
+								existingArgoCD.Spec.SSO.Dex.Resources = defaultArgoCDInstance.Spec.SSO.Dex.Resources
+								changed = true
+							}
+						}
+					}
+				}
+				//lint:ignore SA1019 known to be deprecated
+				if existingArgoCD.Spec.Grafana.Resources == nil { //nolint:staticcheck // SA1019: We must test deprecated fields.
+					//lint:ignore SA1019 known to be deprecated
+					existingArgoCD.Spec.Grafana.Resources = defaultArgoCDInstance.Spec.Grafana.Resources //nolint:staticcheck // SA1019: We must test deprecated fields.
+					changed = true
+				}
+				if existingArgoCD.Spec.HA.Resources == nil {
+					existingArgoCD.Spec.HA.Resources = defaultArgoCDInstance.Spec.HA.Resources
+					changed = true
+				}
+				if existingArgoCD.Spec.Redis.Resources == nil {
+					existingArgoCD.Spec.Redis.Resources = defaultArgoCDInstance.Spec.Redis.Resources
+					changed = true
+				}
+				if existingArgoCD.Spec.Repo.Resources == nil {
+					existingArgoCD.Spec.Repo.Resources = defaultArgoCDInstance.Spec.Repo.Resources
+					changed = true
+				}
+				if existingArgoCD.Spec.Server.Resources == nil {
+					existingArgoCD.Spec.Server.Resources = defaultArgoCDInstance.Spec.Server.Resources
+					changed = true
+				}
+				if defaultArgoCDInstance.Spec.NodePlacement != nil {
+					if !equality.Semantic.DeepEqual(existingArgoCD.Spec.NodePlacement, defaultArgoCDInstance.Spec.NodePlacement) {
+						existingArgoCD.Spec.NodePlacement = defaultArgoCDInstance.Spec.NodePlacement
+						changed = true
+					}
+				} else if existingArgoCD.Spec.NodePlacement != nil {
+					existingArgoCD.Spec.NodePlacement = defaultArgoCDInstance.Spec.NodePlacement
+					changed = true
+				}
+				if !changed {
+					return nil
+				}
+				return r.Client.Update(context.TODO(), existingArgoCD)
+			})
```

Comment:
Can you recheck this logic? Seems like similar checks exists here:
https://github.com/redhat-developer/gitops-operator/blob/9be8c92f5084dccd5801af7c936d4fcd1960015c/controllers/gitopsservice_controller.go#L470-L530